addons/mail/static/src/components/attachment_delete_confirm_dialog/attachment_delete_confirm_dialog.js

```diff
@@ -0,0 +1,88 @@
+odoo.define('mail/static/src/components/attachment_delete_confirm_dialog/attachment_delete_confirm_dialog.js', function (require) {
+'use strict';
+
+const useStore = require('mail/static/src/component_hooks/use_store/use_store.js');
+
+const Dialog = require('web.OwlDialog');
+
+const { Component } = owl;
+const { useRef } = owl.hooks;
+
+class AttachmentDeleteConfirmDialog extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const attachment = this.env.models['mail.attachment'].get(props.attachmentLocalId);
+            return {
+                attachment: attachment ? attachment.__state : undefined,
+            };
+        });
+        // to manually trigger the dialog close event
+        this._dialogRef = useRef('dialog');
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @returns {mail.attachment}
+     */
+    get attachment() {
+        return this.env.models['mail.attachment'].get(this.props.attachmentLocalId);
+    }
+
+    /**
+     * @returns {string}
+     */
+    get CONFIRMATION() {
+        return this.env._t("Confirmation");
+    }
+
+    /**
+     * @returns {string}
+     */
+    getText() {
+        return _.str.sprintf(
+            this.env._t("Do you really want to delete '%s'?"),
+            owl.utils.escape(this.attachment.displayName)
+        );
+    }
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onClickCancel() {
+        this._dialogRef.comp._close();
+    }
+
+    /**
+     * @private
+     */
+    _onClickOk() {
+        this._dialogRef.comp._close();
+        this.attachment.remove();
+        this.trigger('o-attachment-removed', { attachmentLocalId: this.props.attachmentLocalId });
+    }
+
+}
+
+Object.assign(AttachmentDeleteConfirmDialog, {
+    components: { Dialog },
```

Comment:
Follow our convention with `const components = {}` in the requirements above.

Reply:
I used the same pattern as moderation_discard_dialog.js, so should we change all those using this pattern ?

Reply:
Ok, fast and safe change. All comments of this PR just concerns small code style, so after them being applied we'll merge PR right away.